Content.Server/Projectiles/ProjectileSystem.cs

```diff
@@ -77,4 +86,27 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
             RaiseNetworkEvent(new ImpactEffectEvent(component.ImpactEffect, GetNetCoordinates(xform.Coordinates)), Filter.Pvs(xform.Coordinates, entityMan: EntityManager));
         }
     }
+
+    // Remove any embedded objects in the food as it's eaten.
+    private void OnBeforeFullyEaten(EntityUid uid, FoodComponent c, BeforeFullyEatenEvent args)
+    {
+        if (args.Cancelled)
+        {
+            return;
+        }
+
+        UnEmbedChildren(uid);
+    }
+
+    private void UnEmbedChildren(EntityUid embeddee)
+    {
+        var children = Transform(embeddee).ChildEnumerator;
+        while (children.MoveNext(out var child))
+        {
+            if (TryComp<EmbeddableProjectileComponent>(child, out var component))
+            {
+                UnEmbed((child, component), null);
+            }
```

Comment:
I'm not sure how embeddable projectiles work, but at a glance this needs to actually check that the embeddable projectile was embedded in the target, and isn't just stored in a bag or something like that.

Also, while changing that its probably a good idea to give the parent entity some component to denote that it has entities embeded within it, and then use that for the event subscriptions, instead of the current destructible, food, and body subscriptions